src/lib/logical_query_plan/join_node.hpp

```diff
@@ -38,11 +38,34 @@ class JoinNode : public EnableMakeForLQPNode<JoinNode>, public AbstractLQPNode {
    */
   std::shared_ptr<LQPUniqueConstraints> unique_constraints() const override;
 
+  /**
+   * (a) Semi- & Anti-Joins: Forwards left input node's non-trivial FDs
+   * (b) Inner-/Outer-/Cross-Joins:
+   *      - Forwards non-trivial FDs from both input nodes that stay valid.
+   *      - Returns derived, non-trivial FDs from input unique constraints that become discarded due to the join.
```

Comment:
Can you clarify "that stay valid"? Do you mean that they are not nullable?
Also, turn the second item into "Previously trivial FDs turn into non-trivial FDs as the determinant columns become non-unique, but the FDs still hold true"?

Reply:
> Can you clarify "that stay valid"? Do you mean that they are not nullable?

Yes. They become invalid as soon as one determinant expression becomes nullable. Is it necessary to have this level of detail in the function description?

> Also, turn the second item into "Previously trivial FDs turn into non-trivial FDs as the determinant columns become non-unique, but the FDs still hold true"?

Not exactly: If any determinant column becomes non-nullable, the FD is discarded. The uniqueness of determinant expressions does not impact the validity of FDs. 

What about the following?

```cpp
 /**
   * (a) Semi- & Anti-Joins: 
   *      - Forwards left input node's non-trivial FDs
   * (b) Cross-Joins: 
   *      - Forwards non-trivial FDs from both input nodes.
   * (c) Inner-/Outer-Joins:
   *      - Forwards non-trivial FDs from both input nodes whose determinant expressions stay non-nullable.
   *      - Returns derived, trivial FDs from the left and/or right input node as non-trivial FDs if associated unique 
   *        constraints become discarded.
   */
  std::vector<FunctionalDependency> non_trivial_functional_dependencies() const override;
```

Reply:
Sounds good. Minor suggestion for the last point:

Turns derived, trivial FDs from the left and/or right input node into non-trivial FDs if the underlying unique constraints do not survive the join.